### PR TITLE
runtime: Fix virtiofs fd leak

### PR DESCRIPTION
**Pull request by fengwang666**
The kata runtime invokes removeStaleVirtiofsShareMounts after a container is stopped to clean up the stale virtiofs file caches.

Fixes: #6455

### Implementation notes
In my original implementation, I invoked the `RemoveStaleVirtiofsShareMounts` rpc call inside the [UnshareFile](https://github.com/kata-containers/kata-containers/blob/9794c52c6517badf2e3b6f665883d11b37a3b62e/src/runtime/virtcontainers/fs_share_linux.go#L296) method in the fs_share_linux.go and pass in the file path in the rpc call. However turns out this implementation is not very robust because it appears there might be a race where the file inside the guest doesn't expire yet when the `RemoveStaleVirtiofsShareMounts` request arrives. Therefore I change the implementation to scan through all the mounts starting with "/run/kata-containers/shared/containers/" and run `stat` against each of them, which is much more robust against potential delays in virtiofsd.

### CODE REVIEW

**Comment by fengwang666** [comment]
@bergwolf @egernst @gkurz Can you help review this fix?

**Review by liubin — APPROVED** [review]

**Review by amshinde — APPROVED** [review]

File: src/agent/src/rpc.rs [hunk 1]
```diff
@@ -829,6 +831,29 @@ impl agent_ttrpc::AgentService for AgentService {
...
+        is_allowed!(req);
+        let mount_infos = parse_mount_table("/proc/self/mountinfo")
+            .map_err(|e| ttrpc_error!(ttrpc::Code::INTERNAL, e))?;
+        for m in &mount_infos {
+            if m.mount_point.starts_with(KATA_GUEST_SHARE_DIR) {
+                // stat the mount point, virtiofs daemon will remove the stale cache and release the fds if the mount point doesn't exist any more.
+                // More details in https://github.com/kata-containers/kata-containers/issues/6455#issuecomment-1477137277
+                match stat::stat(Path::new(&m.mount_point)) {
+                    Ok(_) => info!(sl!(), "stat {} success", m.mount_point),
+                    Err(e) => info!(sl!(), "stat {} failed: {}", m.mount_point, e),
```
Comment (gkurz) [inline src/agent/src/rpc.rs:849]: Not sure it helps much to say that `stat()` succeeded or failed. We essentially want to know if some stale mounts point are removed. Also should we expect a specific error from virtiofsd in this case ?

Comment (fengwang666) [reply src/agent/src/rpc.rs:849]: @gkurz No. There isn't a specific error returned from virtiofsd. The reason I added the log is to indicate whether a mount is stale or not.

**Pull request merged**